azerothcore/azerothcore-wotlk
# Added emoticons and dialogue events to Warlock Quest NPC Godrick Farsan (ID 5693).

## Changes Proposed:
When the Warlock accepts the mission, "Godrick Farsan" id 5693 summons the target and begins to introduce the target's deeds
[Added video link](https://clipchamp.com/watch/zs6D4hCz0wL)
## Issues Addressed:

- Closes 

## SOURCE:

Blizzard servers Wlk Class Ptr
[Video link](https://clipchamp.com/watch/ei4Yy8feK9A)
## Tests Performed:

- Windows 11
- 


## How to Test the Changes:


1. Create a warlock
2. .go xyz  1782.98990 59.527699 -61.406502 0
3. .level 20
4. .quest add 1472
5. Interact with NPCs Observe whether NPCs start using conversational tasks
6. Complete the mission Continue to interact with the NPC Prepare for "Hearts of the Pure" to see if the NPC starts using 3 emoticons
7. Start the task to see if the NPC's behavior is consistent with the video

## Known Issues and TODO List:


- [ ]The orientation generated by the current NPC is inconsistent with the Blizzard server After the update is just the general direction The actual direction needs to be sniffed from the game


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
